src/main/java/gyro/google/compute/SecurityPolicyRule.java

```diff
@@ -0,0 +1,196 @@
+import gyro.core.resource.Updatable;
+import gyro.core.scope.State;
+import gyro.core.validation.Required;
+import gyro.core.validation.ValidStrings;
+import gyro.google.Copyable;
+
+public class SecurityPolicyRule extends ComputeResource
+    implements Copyable<com.google.api.services.compute.model.SecurityPolicyRule> {
+
+    private String description;
+    private Integer priority;
+    private String action;
+    private Boolean preview;
+    private SecurityPolicyRuleMatcher match;
+
+    /**
+     * The description of the security policy rule.
+     */
+    @Updatable
+    public String getDescription() {
+        return description;
+    }
+
+    public void setDescription(String description) {
+        this.description = description;
+    }
+
+    /**
+     * The priority of the security policy rule.
+     */
+    @Id
+    @Updatable
+    public Integer getPriority() {
+        return priority;
+    }
+
+    public void setPriority(Integer priority) {
+        this.priority = priority;
+    }
+
+    /**
+     * The action to take for this rule. (Required)
+     */
+    @Updatable
+    @Required
+    @ValidStrings({ "allow", "deny(403)", "deny(404)", "deny(502)" })
+    public String getAction() {
+        return action;
+    }
+
+    public void setAction(String action) {
+        this.action = action;
+    }
+
+    /**
+     * The preview flag indicates that this rule is not enforced.
+     */
+    @Updatable
+    public Boolean getPreview() {
+        return preview;
+    }
+
+    public void setPreview(Boolean preview) {
+        this.preview = preview;
+    }
+
+    /**
+     * The match for this rule. (Required).
+     */
+    @Updatable
+    @Required
+    public SecurityPolicyRuleMatcher getMatch() {
+        return match;
+    }
+
+    public void setMatch(SecurityPolicyRuleMatcher match) {
+        this.match = match;
+    }
+
+    @Override
+    public String primaryKey() {
+        return "security-policy-rule-" + getPriority();
+    }
+
+    com.google.api.services.compute.model.SecurityPolicyRule toSecurityPolicyRule() {
+        com.google.api.services.compute.model.SecurityPolicyRule policyRule = new com.google.api.services.compute.model.SecurityPolicyRule();
+        policyRule.setAction(getAction());
+        policyRule.setDescription(getDescription());
+        policyRule.setPriority(getPriority());
+        policyRule.setPreview(getPreview());
+
+        if (getMatch() != null) {
+            policyRule.setMatch(match.toSecurityPolicyRuleMatcher());
+        }
+
+        return policyRule;
+    }
+
+    @Override
+    public void copyFrom(com.google.api.services.compute.model.SecurityPolicyRule securityPolicyRule) {
+        setPriority(securityPolicyRule.getPriority());
+        setDescription(securityPolicyRule.getDescription());
+        setAction(securityPolicyRule.getAction());
+        setPreview(securityPolicyRule.getPreview());
+
+        if (securityPolicyRule.getMatch() != null) {
```

Comment:
So it can reset to null if removed from the console.
```suggestion
        setMatch(null);
        if (securityPolicyRule.getMatch() != null) {
```